### PR TITLE
Fix/support

### PR DESCRIPTION
**Pull request by SamBorisov**
Show live data when we reply, we need to refresh the page now

Need to take a ticket before replying and being able to write (so no multiple admins type a reply at the same time)

Not ‘Close ticket’ -> ‘End Ticket’ ->Could be easily mistaken, also should be with a different color

Long messages for ticket descriptions are not showing well

Re-take possibility (steal ticket)

Make it look better

View button, not “view & reply”

Filter by taken, by your taken tickets, and by subject

Search option - nice to have

Show to who is assigned (by who is taken) when we view it!

On the right is you, and on the left is the player on the chat - better visual

### CODE REVIEW

File: src/services/support/index.js [hunk 1]
```diff
@@ -1,71 +1,100 @@
...
 
-export const sendReplyToTicket = async (id, reply, isTaken) => {
+export const getTicket = async (id) => {
+  const api = useAxios();
+  try {
+    const result = await api.get(`/support-message/ticket/${id}`);
+    return result.data;
+  } catch (err) {
+    console.error(err);
+    throw new Error(err.message);
```
Comment (S-Hristov) [inline src/services/support/index.js:72]: Either use try catch and handle the exception or just let the error fire it is unnecessary to catch an error and throw it agait

File: src/services/support/index.js [hunk 2]
```diff
@@ -1,71 +1,100 @@
...
   } catch (err) {
-    console.log(err);
-    return {
-      data: {
-        action: '',
-        object: ''
-      }
-    };
+    console.error(err);
+    throw new Error(err.message);
```
Comment (S-Hristov) [inline src/services/support/index.js:85]: Either use try catch and handle the exception or just let the error fire it is unnecessary to catch an error and throw it agait

File: src/services/support/index.js [hunk 3]
```diff
@@ -1,71 +1,100 @@
...
```
Comment (S-Hristov) [inline src/services/support/index.js:97]: Either use try catch and handle the exception or just let the error fire it is unnecessary to catch an error and throw it agait

File: src/services/support/index.js [hunk 4]
```diff
@@ -76,12 +105,19 @@ export const retakeTicket = async (id) => {
...
+
+export const closeTicket = async (id) => {
+  const api = useAxios();
+  try {
+    return await api.post(`/support-message/close-ticket`, {
+      MessageId: id,
+    });
+  } catch (err) {
+    console.error(err);
+    throw new Error(err.message);
```
Comment (S-Hristov) [inline src/services/support/index.js:121]: Either use try catch and handle the exception or just let the error fire it is unnecessary to catch an error and throw it agait

File: src/services/support/index.js [hunk 5]
```diff
@@ -76,12 +105,19 @@ export const retakeTicket = async (id) => {
...
   } catch (err) {
-    console.log(err);
-    return {
-      data: {
-        action: '',
-        object: ''
-      }
-    };
+    console.error(err);
+    throw new Error(err.message);
```
Comment (S-Hristov) [inline src/services/support/index.js:109]: Either use try catch and handle the exception or just let the error fire it is unnecessary to catch an error and throw it agait

File: src/services/support/index.js [hunk 6]
```diff
@@ -1,71 +1,100 @@
...
   try {
-    const result = await api.get(`/support-message/tickets`, {});
+    const params = {
+      limit: limit,
+      page: page,
+      sortBy: 'createdAt:DESC'
+    };
 
-    const data = result.data.map((ticket) => {
+    if (Object.keys(filters).length) {
```
Comment (S-Hristov) [inline src/services/support/index.js:12]: Whati s the point of using this just rename the parameters to match the expected payload and let the backend handle it.

**Review by S-Hristov — CHANGES_REQUESTED** [review]

File: src/services/support/index.js (hunk 6 above)
Comment (SamBorisov) [reply src/services/support/index.js]: Made filters with the same name

File: src/services/support/index.js (hunk 1 above)
Comment (SamBorisov) [reply src/services/support/index.js]: no not using try-catch in both service and when we call the function

File: src/services/support/index.js (hunk 2 above)
Comment (SamBorisov) [reply src/services/support/index.js]: no not using try-catch in both service and when we call the function

File: src/services/support/index.js (hunk 3 above)
Comment (SamBorisov) [reply src/services/support/index.js]: no not using try-catch in both service and when we call the function

File: src/services/support/index.js (hunk 4 above)
Comment (SamBorisov) [reply src/services/support/index.js]: no not using try-catch in both service and when we call the function

File: src/services/support/index.js (hunk 5 above)
Comment (SamBorisov) [reply src/services/support/index.js]: no not using try-catch in both service and when we call the function

**Pull request merged**